smartsim/_core/_install/buildenv.py

```diff
@@ -312,18 +345,39 @@ def as_dict(self, db_name: DbEngine = "REDIS") -> t.Dict[str, t.Any]:
         vers = {"Packages": packages, "Versions": versions}
         return vers
 
-    def ml_extras_required(self) -> t.List[str]:
+    def ml_extras_required(self) -> t.Dict[str, t.List[str]]:
         """Optional ML/DL dependencies we suggest for the user.
 
         The defaults are based on the RedisAI version
         """
         ml_defaults = self.REDISAI.get_defaults()
 
-        # remove torch-related fields as they will be installed
-        # by the cli process for use in the RAI build. We don't install
-        # them here as the user needs to decide between GPU/CPU. All other
-        # libraries work on both devices. The correct versions and suffixes
-        # were scraped from https://pytorch.org/get-started/previous-versions/
+        def _format_custom_linux_torch_deps(
+            torchv: str, torchvisionv: str, arc: str
+        ) -> t.Tuple[str, ...]:
+            # The correct versions and suffixes were scraped from
+            # https://pytorch.org/get-started/previous-versions/
+            supported_py_versions = ("3.8", "3.9", "3.10")
+            return tuple(
+                itertools.chain.from_iterable(
+                    (
+                        "torch"
```

Comment:
I think it does! Plus it might be nicer to keep this isolated on the change that the links change, as opposed to needing to dive into some obscure `Versioner` method